pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/inv/OffHeapBitmapInvertedIndexCreator.java

```diff
@@ -71,6 +71,11 @@
   private static final Logger LOGGER =
       LoggerFactory.getLogger(OffHeapBitmapInvertedIndexCreator.class);
 
+  // Number of bytes required to store an integer
+  private static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
+  // Maximum number of entries for each buffer
+  public static final int MAX_NUM_ENTRIES = Integer.MAX_VALUE / INT_SIZE;
```

Comment:
This is not accounting the size correctly. Each entry takes Integer.size + bitmap serialization size. Bitmap serialization size is unpredictable but it's atleast 1 byte but most likely more. So, say, atleast 5 bytes per entry.
So, max entries can not exceed (Integer.max_value / (int_size + 1) - 1). For -1 see class comment